train_models.py

```diff
@@ -21,99 +22,69 @@ def __call__(self, parser, namespace, values, option_string=None):
 def parse_arguments():
     parser = argparse.ArgumentParser()
     parser.register("action", "my_extend", ExtendAction)
-    parser.add_argument("--train", dest="train", action="store", help="The path to train dataset")
     parser.add_argument(
         "--docs",
         dest="docs",
         action="my_extend",
         nargs="+",
         help="Paths to preprocessed docs",
     )
-    parser.add_argument(
-        "-e",
-        dest="epochs",
-        action="store",
-        type=int,
-        help="The number of epochs for training saved_models",
-    )
-    parser.add_argument(
-        "--vs",
-        dest="vector_size",
-        action="store",
-        type=int,
-        default=300,
-        help="Embedding vector size",
-    )
-    parser.add_argument(
-        "--min_count",
-        dest="min_count",
-        action="store",
-        type=int,
-        default=1,
-        help="Ignore all words with total frequency lower than this in W2V, Fasttext and Random models",
-    )
-    parser.add_argument(
-        "--tmp_file",
-        dest="tmp_file",
-        action="store",
-        default=get_tmpfile("pretrained_vectors.txt"),
-        help="The path to tmp file to store pretrained embeddings",
-    )
-    parser.add_argument(
-        "--batch_size",
-        dest="batch_size",
-        action="store",
-        type=int,
-        default=16,
-        help="Batch Size for Bert Model training.",
-    )
     parser.add_argument("--w2v", dest="w2v", action="store_true", help="Train and save word2vec model")
     parser.add_argument("--fasttext", dest="fasttext", action="store_true", help="Train and save fasttext model")
     parser.add_argument("--bert", dest="bert", action="store_true", help="Train and save BERT model for MLM task.")
     parser.add_argument("--sbert", dest="sbert", action="store_true", help="Train and save SBERT model for STS tasks.")
-    parser.add_argument(
-        "--n_examples",
-        dest="n_examples",
-        action="store",
-        type=int,
-        help="Number of bug reports pairs for SBERT train for STS task",
-    )
     return parser.parse_args()
 
 
 def main():
     args = parse_arguments()
+    config = OmegaConf.load(CONFIG_PATH)
 
-    train = pd.read_csv(args.train)
+    train = pd.read_csv(config.datasets.train)
     train_corpus = get_corpus(train)
     docs_corpus = get_docs_text(args.docs)
 
     if args.w2v:
+        cnf_w2v = config.models.w2v
         model = W2VModel(
-            vector_size=args.vector_size, epochs=args.epochs, min_count=args.min_count, tmp_file=args.tmp_file
+            vector_size=cnf_w2v.vector_size,
+            epochs=cnf_w2v.epochs,
+            min_count=cnf_w2v.min_count,
+            pretrained_model=cnf_w2v.pretrained,
+            tmp_file=cnf_w2v.tmp_file or get_tmpfile("pretrained_vectors.txt"),
         )
         model.train_and_save_all(train_corpus, docs_corpus)
```

Comment:
You can avoid most of the copy-pasting by using config in the models instead of parsing it here. Then you can just select the needed class based on a stored mapping and pass config there. It will make the code more concise.

Reply:
Now it looks like this: 
```python
model = W2VModel(**config.models.w2v)
```